spec/index.html

```diff
@@ -839,32 +841,149 @@ <h3>Terms and Formulae</h3>
+        <li>M((e<sub>1</sub>... e<sub>n</sub>))=(M(e<sub>1</sub>)... M(e<sub>n</sub>)) for all lists  e=(e<sub>1</sub>... e<sub>n</sub>) which are nodes of G.</li>
+        <li>The triple ( s, p, o ) is in G if and only if the triple ( M(s), M(p), M(o) ) is in G'.</li>
+        <li>M({F})={F'} where the nested graph F is isomorphic to F'.</li>
+      </ol>
+      We write G&equiv;G'.
       </p>
+
+
       </section>
 
 
 
 
       <section id='sinterpret'>
         <h3>Simple N3 interpretations</h3>
+      <p>
+      A <dfn data-lt="simpleI">simple N3 interpretation</dfn> is a structure I consisting of:
+      <ol>
+        <li> A non-empty set IR of resources, called the domain or universe of I.</li>
+        <li> A set IP, called the set of properties of I.</li>
+        <li> A mapping IEXT from IP into the powerset of IR x IR i.e. the set of sets of pairs < x, y > with x and y in IR.</li>
+        <li> A mapping IS from IRIs into (IR union IP). </li>
+        <li> A partial mapping IL from literals into (IR union IP). </li>
+        <li> A mapping IG from closed graphs into (IR union IP), such that IG(E)=IG(F) for all closed graphs E&equiv;F. </li>
+        </ol>
+        IEXT(x), called the extension of x, is a set of pairs which identify the arguments for which the property is true, that is, a binary relational extension.
+        The distinction between IR and IL will become significant below when the semantics of datatypes are defined. IL is allowed to be partial because some literals may fail to have a referent.
+      </p>
+      <p>
+        The denotation of a closed N3 graph in a simple interpretation I is then given by the following rules,
+        where the interpretation is also treated as a function from expressions (names, triples and graphs) to elements of the universe and truth values:
+      </p>
+      <p>
+        <dfn data-lt="semantics">Semantic conditions for closed graphs</dfn>
+        <ul>
+        <li>if E is a literal then I(E) = IL(E)</li>
+        <li>if E is an IRI then I(E) = IS(E)</li>
+        <li>if E is a universal free quoted graph then I(E)=IG(E) and IG(E)&equiv;E</li>
+        <li>if E is a closed triple s p o. with p	&#8800; log:implies, or s&notin;QG&cup;{true,false} or o&notin;QG&cup;{true,false}, then I(E) = true if</li>
+          <ul>
+        <li>I(p) is in IP and the pair <I(s),I(o)> is in IEXT(I(p))</li>
+
+            <li>otherwise I(E) = false.</li>
+        </ul>
+        <li>if E is a triple of the form {f} log:implies {g}. then E is true if from I(f)=true follows that I(g)=true. </li>
+        <li>if E is a triple of the form {f} log:implies false. then E is true if I(f)=false. </li>
+          <li>if E is a triple of the form true log:implies {g}. then E is true if I(g)=true. </li>
+          <li>if E is a triple of the form {f} log:implies true. or false log:implies {g}. then I(E)=true.</li>
+          <li> if E=true log:implies false. then I(E)=false., </li>
```

Comment:
Couldn't we simplify all this by introducing a new function IC (C for "clause"), such that
- If E = { f } is a quoted formula, IC(E) = I(E)
- If E = b a boolean, IC(b) = b
- otherwise, IC(E) is not defined

Then I(C1 => C2) would be true if IC(C1) implies IC(C2)?

---
Also, in these lines, a number of `then E is true` should be `then I(E)=true`

Reply:
You are right that the handling of the implication should get less ugly, I will fix it as suggested.

The mistake with I(E) is already corrected.